### PR TITLE
Fix: Metadata ignored for non model-specific EditorTemplate

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #2778 

@dougbu

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.FunctionalTests/HtmlGenerationTest.cs [hunk 1]
```diff
@@ -539,5 +539,39 @@ public class HtmlGenerationTest
...
+            var expected = "<label class=\"control-label col-md-2\" for=\"Name\">ItemName</label>\r\n" +
+                "<input id=\"Name\" name=\"Name\" type=\"text\" value=\"\" />\r\n\r\n" +
+                "<label class=\"control-label col-md-2\" for=\"Id\">ItemNo</label>\r\n" +
+                "<input data-val=\"true\" data-val-required=\"The ItemNo field is required.\" id=\"Id\" name=\"Id\" type=\"text\" value=\"\" />\r\n\r\n";
+
+            // Act
+            var response = await client.GetStringAsync("http://localhost/HtmlGeneration_Home/ItemUsingSharedEditorTemplate");
+
+            // Assert
+            Assert.Equal(expected, response);
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/HtmlGenerationTest.cs]: Need `ignoreLineEndingDifferences: true` here and in next test. This is the reason Travis is currently failing.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/HtmlGenerationTest.cs [hunk 2]
```diff
@@ -539,5 +539,39 @@ public class HtmlGenerationTest
...
+        [Fact]
+        public async Task EditorTemplateWithNoModel_RendersWithCorrectMetadata()
+        {
+            // Arrange
+            var server = TestHelper.CreateServer(_app, SiteName, _configureServices);
+            var client = server.CreateClient();
+            var expected = "<label class=\"control-label col-md-2\" for=\"Name\">ItemName</label>\r\n" +
+                "<input id=\"Name\" name=\"Name\" type=\"text\" value=\"\" />\r\n\r\n" +
+                "<label class=\"control-label col-md-2\" for=\"Id\">ItemNo</label>\r\n" +
+                "<input data-val=\"true\" data-val-required=\"The ItemNo field is required.\" id=\"Id\" name=\"Id\" type=\"text\" value=\"\" />\r\n\r\n";
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/HtmlGenerationTest.cs]: Don't write "\r\n". Instead use a verbatim `string` literal; that will be easier to read, much more consistent with other tests, and platform-neutral.

File: test/WebSites/HtmlGenerationWebSite/Views/Shared/EditorTemplates/Item.cshtml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+﻿@using HtmlGenerationWebSite.Models
+@model Item
+@Html.Label(nameof(Model.Description))
+@Html.TextBoxFor(m => m.Description)
```
Comment (dougbu) [inline test/WebSites/HtmlGenerationWebSite/Views/Shared/EditorTemplates/Item.cshtml]: To match the shared case more closely, add a `String.cshtml` template containing `@model string` plus the lines in `Common.cshtml`. The expectations shouldn't need to change...

File: src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Internal/TemplateBuilder.cs [hunk 4]
```diff
@@ -72,7 +72,7 @@ public string Build()
             // We need to copy the ModelExplorer to copy the model metadata. Otherwise we might
             // lose track of the model type/property. Passing null here explicitly, because
             // this might be a typed VDD, and the model value might not be compatible.
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Internal/TemplateBuilder.cs]: Add a bit to this comment about why the next line creates a `ViewDataDictionary<object>`.

**Comment by dougbu** [comment]
:watch:

**Comment by dougbu** [comment]
Also please add a test or two to `RazorPageActivatorTest` covering expected `RazorPage.ViewData` values when activation context contains a `ViewDataDictionary<object>` instance.

File: src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Internal/TemplateBuilder.cs (hunk 4 above)
Comment (ajaybhargavb) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Internal/TemplateBuilder.cs]: Added.

**Comment by ajaybhargavb** [comment]
Updated.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/HtmlGenerationTest.cs [hunk 5]
```diff
@@ -539,5 +539,42 @@ public class HtmlGenerationTest
...
+            Assert.Equal(expected, response);
+        }
+
+        [Fact]
+        public async Task EditorTemplateWithSpecificModel_RendersWithCorrectMetadata()
+        {
+            // Arrange
+            var server = TestHelper.CreateServer(_app, SiteName, _configureServices);
+            var client = server.CreateClient();
+            var expected = PlatformNormalizer.NormalizeContent(
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/HtmlGenerationTest.cs]: Should not need `PlatformNormalizer.NormalizeContent()` here.

**Comment by dougbu** [comment]
:shipit: after removing extra `NormalizeContent` and squashing

**Comment by rynowak** [comment]
:shipit: from me

**Pull request closed without merging**